python/lsst/resources/gs.py

```diff
@@ -262,7 +264,7 @@ def open(
         *,
         encoding: Optional[str] = None,
         prefer_file_temporary: bool = False,
-    ) -> Iterator[IO]:
+    ) -> Iterator[ResourceHandleProtocol]:
```

Comment:
hmm, my type checkers are not throwing anything, but I suppose it is because it is hidden under so many layers and ultimately import guards, this may take a bit to sort out, or I will return a Union

Reply:
Actually, it looks like what is returned from the google code is either `TextIoWrappers` or a subclass of `io.BufferedIOBase`, which should be ok with the protocol